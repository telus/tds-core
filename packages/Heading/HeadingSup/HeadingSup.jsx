import React from 'react'
import PropTypes from 'prop-types'
import safeRest from '../../../shared/utils/safeRest'

import styles from '../Heading.modules.scss'

/**
 * Superscript text for `Heading` an as HTML `<sup>` element.
 *
 * _This component can only be accessed as a name-spaced component: `Heading.Sup`._
 */
const HeadingSup = ({ children, ...rest }) => (
  <sup {...safeRest(rest)} className={styles.sup}>
    {children}
  </sup>
)

HeadingSup.propTypes = {
  /**
   * The text.
   */
  children: PropTypes.string.isRequired,
}

HeadingSup.displayName = 'Heading.Sup'

export default HeadingSup
