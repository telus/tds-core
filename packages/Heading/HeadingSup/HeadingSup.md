```
<Box between={3}>
  <Heading level="h1">TELUS Pik TV<Heading.Sup>&trade;</Heading.Sup></Heading>
  <Heading level="h2">TELUS Pik TV<Heading.Sup>&trade;</Heading.Sup></Heading>
  <Heading level="h3">TELUS Pik TV<Heading.Sup>&trade;</Heading.Sup></Heading>
  <Heading level="h4">TELUS Pik TV<Heading.Sup>&trade;</Heading.Sup></Heading>
</Box>
```
