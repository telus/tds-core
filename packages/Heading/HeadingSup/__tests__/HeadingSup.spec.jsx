import React from 'react'
import { shallow } from 'enzyme'

import HeadingSup from '../HeadingSup'

describe('HeadingSup', () => {
  const doShallow = props => shallow(<HeadingSup {...props}>Some content</HeadingSup>)

  it('renders', () => {
    const headingSup = doShallow()

    expect(headingSup).toMatchSnapshot()
  })

  it('renders an HTML sub tag', () => {
    const headingSup = doShallow()

    expect(headingSup).toHaveDisplayName('sup')
  })

  it('passes additional attributes to the sup element', () => {
    const headingSup = doShallow({ id: 'the-sup-text' })

    expect(headingSup).toHaveProp('id', 'the-sup-text')
  })

  it('does not allow custom CSS', () => {
    const headingSup = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(headingSup).not.toHaveProp('className', 'my-custom-class')
    expect(headingSup).not.toHaveProp('style')
  })
})
