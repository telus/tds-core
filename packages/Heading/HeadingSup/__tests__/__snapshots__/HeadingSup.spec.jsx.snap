// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingSup renders 1`] = `
<sup
  className="sup"
>
  Some content
</sup>
`;
